icu4c/source/common/loclikely.cpp

```diff
@@ -464,8 +464,22 @@ parseTagString(
         goto error;
     }
 
-    subtagLength = ulocimp_getLanguage(position, lang, *langLength, &position);
-    u_terminateChars(lang, *langLength, subtagLength, err);
+    {
+        icu::CharString result = ulocimp_getLanguage(position, &position, *err);
+        if (U_FAILURE(*err)) {
+            goto error;
+        }
+
+        subtagLength = result.length();
+        if (subtagLength > *langLength) {
+            *lang = '\0';
```

Comment:
This should not be necessary.

The following should really suffice, as suggested before:
```
if (subtagLength <= *langLength) {
    uprv_memcpy(lang, result.data(), subtagLength);
}
```

The already then-following u_terminateChars() will set the buffer overflow error, and the next statement takes care of the `goto error;`.

Keep it simple.

Reply:
Done.